app/src/main/java/com/github/onlynotesswent/ui/deck/DeckMenu.kt

```diff
@@ -0,0 +1,521 @@
+                  text = { Text("Add new card") },
+                  icon = { Icon(imageVector = Icons.Default.Add, contentDescription = "Add card") },
+                  onClick = onAddCardClick,
+                  modifier = Modifier.testTag("addCardMenuItem")),
+              CustomDropDownMenuItem(
+                  text = { Text("Import deck") },
+                  icon = {
+                    Icon(imageVector = Icons.Default.Download, contentDescription = "Import deck")
+                  },
+                  onClick = onImportDeckClick,
+                  modifier = Modifier.testTag("importDeckMenuItem")),
+              CustomDropDownMenuItem(
+                  text = { Text("Edit Deck") },
+                  icon = {
+                    Icon(imageVector = Icons.Default.Edit, contentDescription = "Edit deck")
+                  },
+                  onClick = onEditClick,
+                  modifier = Modifier.testTag("editDeckMenuItem"))),
+      expanded = expandedDropdownMenu.value,
+      onFabClick = { expandedDropdownMenu.value = !expandedDropdownMenu.value },
+      onDismissRequest = { expandedDropdownMenu.value = false },
+      fabIcon = { Icon(imageVector = Icons.Default.Edit, contentDescription = "Edit Deck") })
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+// Modularised functions for the top app bar, deck info, sort options and play modes
+
+/**
+ * Composable function that represents the top app bar for the Deck Menu screen.
+ *
+ * @param onBackButtonClick The callback to be invoked when the back button is clicked.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun DeckMenuTopAppBar(onBackButtonClick: () -> Unit) {
+  ScreenTopBar(
+      title = "Deck",
+      titleTestTag = "deckTopBarTitle",
+      onBackClick = onBackButtonClick,
+      icon = {
+        Icon(imageVector = Icons.AutoMirrored.Outlined.ArrowBack, contentDescription = "Back")
+      },
+      iconTestTag = "backButton")
+}
+
+/**
+ * Composable function that displays the card count and author information for a deck.
+ *
+ * @param deck The state containing the deck information.
+ * @param author The state containing the author information.
+ * @param userViewModel The ViewModel for user-related data.
+ * @param fileViewModel The ViewModel for file-related data.
+ * @param navigationActions The actions for navigation.
+ */
+@Composable
+private fun CardCountAndAuthor(
+    deck: State<Deck?>,
+    author: State<User?>,
+    userViewModel: UserViewModel,
+    fileViewModel: FileViewModel,
+    navigationActions: NavigationActions
+) {
+  Row(
+      verticalAlignment = Alignment.CenterVertically,
+      horizontalArrangement = Arrangement.spacedBy(10.dp)) {
+        deck.value?.let {
+          Text(
+              text =
+                  it.flashcardIds.size.let { count ->
+                    if (count == 1) "1 card" else "$count cards"
```

Comment:
Nice thinking about plurials in this case, but as my other comments unfortuanly I have to say: "This shoud be modularized with string resources"